### DISCUSSION TITLE
Was man wissen sollte

### DISCUSSION THREAD

**Discussion by dhaude** [tbnobody/OpenDTU#82] [General]
Es ist Sonntag der 14.08.2022 an einem schon recht warmen Vormittag. Ich erfreue mich an OpenDTU, die mir zuverlässig die Werte meiner HM-1500 anzeigt - seit ein paar Tagen in der Version 0.1.19
Mein Schreibtisch musste mal wieder aufgeräumt werden und das USB-Kabel zur OpenDTU lag im Weg.
Also kurz dis-connect vom Netzteil, anders gelegt und wieder den USB-Stecker ins Netzteil gesteckt.
Ich war es ja gewohnt, dass es ein paar Minuten dauert, bis OpenDTU die Verbindung mit dem WR aufnimmt, bis die Werte aktualisiert und angezeigt werden. Aber diesmal nichts. Position geändert, mehrfach RESET ausgeführt und auch der Betrieb über die VSC-Console zeigte keine Fehler - allerdings sah man auch kein Datenverkehr.
Die Infos zeigten auch keine Probleme - bis auf NTP, dort wurde nichts angezeigt und so langsam dämmerte es mir.
Der Telekom Internetzugang war seit einer Stunde gestört und die Jungs brauchten noch weitere 27 Stunden, bis wir wieder online waren. 
Zu diesem Zeitpunkt dann konnte man die OpenDTU neu booten und sie lief wieder wie gewohnt.
Was lernen wir daraus? Schreibtisch nicht mehr aufräumen (never touch a running system)?
Ist das ein bekanntes Verhalten?

**Comment by tbnobody** [comment]
Ja.... ohne aktuelle Uhrzeit keine Abfrage des WR. Jedes Paket das an den WR geschickt wird enthält die aktuelle Zeit. Diese nutzt der WR unter anderem wohl für sein internes Eventlog usw.

**Comment by turrican944** [comment]
Moin
Was lernen wir daraus immer einen eigenen Zeitserver am laufen haben falls das Internet mal ausgefallen ist :-)

**Reply by iot4all** [reply]
Oder eine RTC an den ESP klöppen. Sind noch Pins für I2C-Bus frei?
http://www.esp32learning.com/code/esp32-and-ds3231-rtc-example.php

**Reply by m0tschi** [reply]
Jede Fritzbox bzw. viele Router haben einen eigenen Zeitserver, einfach diesen als NTP Konfigurieren :)

**Comment by VientoFinki** [comment]
Ich habe das Problem, dass ich eine Nulleinspeisung realisiert habe, die ohne Internet in Rumänien laufen soll. Dort fällt öfter der Strom aus und die Zähler sind always positiv Zähler (man zahlt seinen Strompreis für das was man verbraucht und was man einspeist !! irre!!).
Wenn kein Zeitserver vorhanden ist kann man ja die Zeit manuell per Tastendruck synchronisieren. In der Liveansicht nach dem Booten der openDTU erscheint dann der Hinweis. Das funktioniert auch nur kann man dort nicht nach jedem Stromausfall die Zeit synchronisieren.
Lösungen gibt es aber alle brauchen zusätzliche Geräte.
Meine Idee:
Die Manuelle Zeit Synchronisation gleich beim Hochfahren ausführen lassen wie wenn der Button "Zeit synchronisieren" gedrückt worden wäre.  Die Uhrzeit ist dort egal wichtig ist nur dass der MQTT Server läuft und mein Regler auf dem Raspberry die richtigen Werte vom HM-1500 und dem EM3 Shelly einliest das neue Maximum berechnet und auf dem HM-1500 setzt. Was auch alles ohne Internet funktioniert, wenn ich beim Hochstarten mit einer direkten LAN-Verbindung zum Webserver openDTU mit dem Button die Zeit synchronisiere.
Bei meinen bisherigen Software-Änderungen an der openDTU SW konnte ich die Abfrage wegen fehlender Zeitsynchronisation übergehen in dem ich in WebApi_ntp.cpp mal zum Testen dirty die Aufrufe mit getLocalTime immer mit true returnen lasse.

Hat jemand Ideen ob das zum Ziel führt  oder wie man das schnell umsetzen könnte ohne GPS Mausoder andere RTCs ?

**Comment by VientoFinki** [comment]
Ergänzung im Menü  NTP-Informationen erscheint dann auch die Ausgabe dass der Server synchronisiert ist es erfolgt aber kein Aktualisierung der Daten vom HM-1500.
Konfigurationszusammenfassung
Server	192.168.0.1
Zeitzone	CET-1CEST,M3.5.0,M10.5.0/3
Zeitzonenbeschreibung	Europe/Berlin
Aktuelle Zeit
Status	synchronisiert
Lokale Uhrzeit	Thursday, January 01 1970 01:13:43
Morgendämmerung	Thursday, January 01 1970 07:00:00
Abenddämmerung	Thursday, January 01 1970 17:43:00
Modus	Nacht

**Comment by VientoFinki** [comment]
Unglaublich, es war noch die falsche Antenne aufgesteckt aber jetzt  nach der SW-Änderung läuft die Nulleinspeisung ohne Internet-Anschluss völlig autarkt. Die Abfrage auf den NTP wird per SW-Änderung von mir quittiert. Einziges Manko die Zeit startet immer bei 1970 1:0:0. Gerät einfach an irgendeiner Steckdose einstecken und läuft, egal wo, Hauptsache der Hoymiles erkennt das Stromnetz, dann speist er ein und mein Regler unter openHAB regelt nach meinem programmierten Algo. Auch lassen sich per MQTT mit OH4 Werte und Infos wie die IP-Adresse der openDTU anzeigen (auf Handy oder Laptop).Ebenso läßt sich über den TP-Link auch der Webserver der openDTU adressieren und die Werte der Strings und die Energiemengen auslesen. Die notwendigen Geräte zu den Solarmodulen sind ein Wechselrichter von  Hoymiles mit DTU Antenne, ein RPI4, eine openDTU, ein EM3-Shelly und ein TP-Link Router aus dem Sperrmüll mit meiner Programm-Änderung. Ein Gerät das auch in den Ländern mit always positiv Zählern einfach zu installieren, warten und reparieren ist.  Ich hoffe ich mach Mut zu mehr Solar Ernte zum Wohle des Planeten und der Bevölkerung die sehr wenig für Ihre Arbeit entlohnt wird. Es ist meiner Meinung nach aktuell die Preisgünstigste Lösung für die widrigsten Umstände bezüglich Kommunikationstechnik, Stromzähler und den Verträgen im Stromhandel.  Viele sonnige Grüße ohne den Regen zu vernachlässigen.
